dont match if session name is a subset of another session name

kelaban opened the pull request:
Current tmux will match a session name which is a subset of another session name and attach to that session instead of creating a new session.

For example if you try and create a new session name `abc` but there is already a session named `abcd` it will attach to `abcd` instead of creating a new one

dennishafemann commented:
```
"\b${TMUX_SESSION_NAME}\b"
```
... will also find a subset. E.g. -ts "test abc def" and -ts "abc". \b is regexp for word-binding. So, you're right, subsets of the sessionname should'nt be found accidently unless it should be found via the regexp of the session-name... I would prefer a ^ and $ to set explicitly the borders.

kelaban commented:
Updated to use `tmux ls -F '#S'` which prints only the names of the sessions in the output. Also updated the grep command to use `^` and `$` anchors accordingly.
